new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java

```diff
@@ -0,0 +1,387 @@
+   * Delete Operator and install latest Operator and verify CRD version is updated
+   * and the domain can be managed by scaling the cluster using operator REST api.
+   */
+  @Test
+  @DisplayName("Upgrade Operator from 2.6.0 to latest")
+  @MustNotRunInParallel
+  public void testOperatorUpgradeFrom2_6_0(@Namespaces(3) List<String> namespaces) {
+    this.namespaces = namespaces;
+    upgradeOperator("2.6.0", false);
+  }
+
+  /**
+   * Operator upgrade from 3.0.0 to latest.
+   * Install 3.0.0 Operator from GitHub chart repository and create a domain.
+   * Upgrade operator with latest Operator image and verify CRD version and image are updated
+   * and the domain can be managed by scaling the cluster using operator REST api.
+   */
+  @Test
+  @DisplayName("Upgrade Operator from 3.0.0 to latest")
+  @MustNotRunInParallel
+  public void testOperatorUpgradeFrom3_0_0(@Namespaces(3) List<String> namespaces) {
+    this.namespaces = namespaces;
+    upgradeOperator("3.0.0", true);
+  }
+
+  /**
+   * Cleanup Kubernetes artifacts in the namespaces used by the test and
+   * delete CRD.
+   */
+  @AfterEach
+  public void tearDown() {
+    if (System.getenv("SKIP_CLEANUP") == null) {
+      CleanupUtil.cleanup(namespaces);
+      new Command()
+          .withParams(new CommandParams()
+              .command("kubectl delete crd domains.weblogic.oracle --ignore-not-found"))
+          .execute();
+    }
+  }
+
+  private void upgradeOperator(String operatorVersion, boolean useHelmUpgrade) {
+    logger.info("Assign a unique namespace for operator {0}", operatorVersion);
+    assertNotNull(namespaces.get(0), "Namespace is null");
+    final String opNamespace1 = namespaces.get(0);
+    logger.info("Assign a unique namespace for latest operator");
+    assertNotNull(namespaces.get(1), "Namespace is null");
+    final String opNamespace2 = namespaces.get(1);
+    logger.info("Assign a unique namespace for domain");
+    assertNotNull(namespaces.get(2), "Namespace is null");
+    String domainNamespace = namespaces.get(2);
+    latestOperatorImageName = getOperatorImageName();
+
+    // delete existing CRD
+    new Command()
+        .withParams(new CommandParams()
+            .command("kubectl delete crd domains.weblogic.oracle --ignore-not-found"))
+        .execute();
+
+    HelmParams opHelmParams =
+        new HelmParams().releaseName("weblogic-operator")
+            .namespace(opNamespace1)
+            .repoUrl(OPERATOR_GITHUB_CHART_REPO_URL)
+            .repoName("weblogic-operator")
+            .chartName("weblogic-operator")
+            .chartVersion(operatorVersion);
+
+    // install operator
+    String opNamespace = opNamespace1;
+    String opServiceAccount = opNamespace + "-sa";
+    installAndVerifyOperator(opNamespace, opServiceAccount, true,
+        0, opHelmParams, domainNamespace);
+
+    // create domain
+    createDomainHomeInImageAndVerify(domainNamespace, operatorVersion);
+
```

Comment:
If the tests fails here the CRD version will not be latest?

Reply:
If the test fails, CRD is deleted in the afterEach method.

Reply:
ok good.